F02A02 Adventsaufgabe - Polylingual Pingu/Würgeschlangen/Würgeschlangen.py

```diff
@@ -108,22 +108,22 @@ def incubate(self, current_day: int) -> Union[Egg, Snake]:
 
 
 # let Adam and Eve lay 3 eggs
-for i in range(3):
+for _ in range(3):
     terrarium["eggs"].append(adam.breed(eve))
     print("🪺 An egg was laid!")
 
 
 # now incubate the eggs until all the baby snakes hatched
 print("The eggs will now be incubated")
 day = 0
-while not len(terrarium["eggs"]) == 0:
+while terrarium["eggs"]:
     print("☀️ A new day " + str(day) + " begins")
     for i in range(len(terrarium["eggs"])):
         print("Incubating egg in the hatchery station at position", str(i),
               " - needs", str(terrarium["eggs"][i].days_until_hatch), "more days to hatch"
               )
         egg_or_snake = terrarium["eggs"][i].incubate(day)
-        if type(egg_or_snake) == Snake:
+        if isinstance(egg_or_snake, Snake):
```

Comment:
isinstance is preferable to type(x) == X, as it handles inheritance correctly. Here both variants work, but in general isinstance is the best practice.